speaker/bgp_controller.go

```diff
@@ -83,22 +84,50 @@ newPeers:
 }
 
 // nodeHasHealthyEndpoint return true if this node has at least one healthy endpoint.
-func nodeHasHealthyEndpoint(eps *v1.Endpoints, node string) bool {
+func nodeHasHealthyEndpoint(eps k8s.EpsOrSlices, node string) bool {
```

Comment:
I'm wondering could these two functions (and the similar ones in the l2 case) could somehow be combined into one function or multiple functions that lived in the k8s package in a way that could hide the details of Eps versus EpSlices to users of the k8s package.

What do you think @fedepaol ?

Reply:
Not sure I get the comment right. My understanding is, we can have a `k8s.nodeHasHealthyEndpoint(svc, node)` function that abstracts away the endpoints source.
By doing that, we would introduce a dependency of `shouldAnnounce` to the state of the cache which I am not entirely sure I like.
This would make testing a bit harder, and not sure the extra complexity would add corner cases. 

I like the fact that we fetch everything we need in the k8s package and we explicitly pass them to `shouldAnnounce`. 
Said that, if you think it's better the other way around I can rework this.

Reply:
Sorry for not being clear. There seems to be duplicate code in nodeHasHealthyEndpoints(), healthyEndpointExists(), usableNodes() so I was just wondering if there was a way to combine them somehow. However, looking at them again, its mostly looping over endpoints structures so maybe not. What do you think?

Reply:
Ok, if this is about refactoring them and removing duplication, I think there's room for improvement.